src/vs/platform/files/common/files.ts

```diff
@@ -105,6 +105,27 @@ export interface IFileService {
 	del(resource: URI, useTrash?: boolean): TPromise<void>;
 
 	/**
+	 * Backs up the provided file to a temporary directory to be used by the hot
+	 * exit feature and crash recovery.
+	 */
+	backupFile(resource: URI, content: string): TPromise<IFileStat>;
```

Comment:
I meanwhile moved ITextFileService out of file contrib land into the workbench so there was a chance of layer breaking. But now that the service is in workbench core it is on the same level as untitled files.